src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -1991,15 +1991,18 @@ void C2_MacroAssembler::neon_reduce_add_integral(Register dst, BasicType bt,
 
 // Vector reduction multiply for integral type with ASIMD instructions.
 // Note: temporary registers vtmp1 and vtmp2 are not used in some cases.
+// Note: vsrc and vtmp2 may match.
 // Clobbers: rscratch1
-void C2_MacroAssembler::neon_reduce_mul_integral(Register dst, BasicType bt,
-                                                 Register isrc, FloatRegister vsrc,
-                                                 unsigned vector_length_in_bytes,
-                                                 FloatRegister vtmp1, FloatRegister vtmp2) {
+void C2_MacroAssembler::reduce_mul_integral_le128b(Register dst, BasicType bt, Register isrc,
+                                                   FloatRegister vsrc,
+                                                   unsigned vector_length_in_bytes,
+                                                   FloatRegister vtmp1, FloatRegister vtmp2) {
   assert(vector_length_in_bytes == 8 || vector_length_in_bytes == 16, "unsupported");
+  assert_different_registers(vtmp1, vsrc);
+  assert_different_registers(vtmp1, vtmp2);
```

Comment:
nit: would be neat to use 
```suggestion
  assert_different_registers(vsrc, vtmp1, vtmp2);
```